package.json

```diff
@@ -38,30 +38,22 @@
 		"@automattic/muriel-style": "file:./packages/muriel-style",
 		"@automattic/tree-select": "file:./packages/tree-select",
 		"@babel/cli": "7.4.4",
-		"@babel/core": "7.4.4",
-		"@babel/plugin-syntax-dynamic-import": "7.2.0",
 		"@babel/plugin-syntax-jsx": "7.2.0",
```

Comment:
Not sure what to make of these :thinking: For consistency's sake, it might make sense to move them to `calypso-build`, but OTOH, they aren't directly used by it...

Reply:
I see the `@babel/plugin-syntax-jsx` used only in unit test of `packages/babel-plugin-transform-react-jsx`. Everywhere else, it's a transitive dependency of `@babel/preset-react`. And that preset is missing from the root `package.json`.

We can remove it from here and add it as dev dependency of `packages/babel-plugin-transform-react-jsx`, what do you think?

Reply:
I think that this makes sense, in principle :slightly_smiling_face:, and I've pushed a commit to do that (d24faaf3e0).

However, we've had some problems in the past with individual `packages/`' `devDependencies` not being taken into account when running `npm ci` (or was it `lerna`?)

`babel-plugin-transform-react-jsx` seems to be a bit of a special case, since it doesn't have a `build` command (it's a babel plugin, so it has to be CommonJS anyway). The package is also still set to `private`. Finally, as you pointed out, `@babel/plugin-syntax-jsx` is only used for the package's unit tests, so it's not required during Calypso's build process.

Calypso still seems to build and run fine. If there's any problem with the package, I guess we'll find out the hard way when attempting to publish it :sweat_smile: (unless we manage to retire it in favor of `import()` statements and React Suspense anyway) :slightly_smiling_face:

Reply:
The `@babel/plugin-syntax-jsx` dependency can be actually eliminated and I do that in #33161. Not that it would help much: the test still depends on `@babel/core`, which is not declared as a dependency either.

When the `@babel/core` dep is moved from `wp-calypso/package.json` to `wp-calypso/packages/calypso-build/package.json`, it's no longer guaranteed that the package is available for `packages/babel-plugin-transform-react-jsx` in `wp-calypso/node_modules`. I don't know how to solve that. Would adding a `devDependency` help hoist the package to the top?

Reply:
devDependencies on linked / internal packages are ignored by `npm` (and by `lerna` when using `file:` refs to link packages). It's the same as if you installed a package via a git URL. So devDependencies all have to move to the root `package.json` to get installed.